ambry-commons/src/test/java/com.github.ambry.commons/ByteBufferReadableStreamChannelTest.java

```diff
@@ -156,118 +61,60 @@ public void readAndWriteCornerCasesTest() throws Exception {
+   * 2. Create with a null list of {@link ByteBuffer}.
+   * 3. Create with an empty {@link ByteBuffer} list.
+   * 4. Create with a list of {@link ByteBuffer} containing a null {@link ByteBuffer}.
    */
   @Test
-  public void idempotentOperationsTest() throws IOException {
-    byte[] in = fillRandomBytes(new byte[1]);
-    ByteBufferReadableStreamChannel byteBufferReadableStreamChannel =
-        new ByteBufferReadableStreamChannel(ByteBuffer.wrap(in));
-    assertTrue("ByteBufferReadableStreamChannel is not open", byteBufferReadableStreamChannel.isOpen());
-    byteBufferReadableStreamChannel.close();
-    assertFalse("ByteBufferReadableStreamChannel is not closed", byteBufferReadableStreamChannel.isOpen());
-    // should not throw exception.
-    byteBufferReadableStreamChannel.close();
-    assertFalse("ByteBufferReadableStreamChannel is not closed", byteBufferReadableStreamChannel.isOpen());
-  }
-
-  // helpers
-  // general
-
-  /**
-   * Fills random bytes into {@code in}.
-   * @param in the byte array that needs to be filled with random bytes.
-   * @return {@code in} filled with random bytes.
-   */
-  private byte[] fillRandomBytes(byte[] in) {
-    new Random().nextBytes(in);
-    return in;
-  }
-}
+  public void constructionErrorCasesTest() {
+    // null byteBuffer
+    try {
+      new ByteBufferReadableStreamChannel((ByteBuffer) null);
+      fail("Construction of ByteBufferReadableStreamChannel should have failed");
+    } catch (RuntimeException e) {
+      assertEquals("Unexpected RuntimeException", NullPointerException.class, e.getClass());
+    }
 
-/**
- * Callback for read operations on {@link ByteBufferReadableStreamChannel}.
- */
-class ReadIntoCallback implements Callback<Long> {
-  public volatile long bytesRead;
-  public volatile Exception exception;
-  private final AtomicBoolean callbackInvoked = new AtomicBoolean(false);
-  private final CountDownLatch latch = new CountDownLatch(1);
+    // null bufferList
+    constructionFailureTest(null, IllegalArgumentException.class);
 
-  @Override
-  public void onCompletion(Long result, Exception exception) {
-    if (callbackInvoked.compareAndSet(false, true)) {
-      bytesRead = result;
-      this.exception = exception;
-      latch.countDown();
-    } else {
-      this.exception = new IllegalStateException("Callback invoked more than once");
-    }
-  }
+    // empty bufferList
+    constructionFailureTest(Collections.EMPTY_LIST, IllegalArgumentException.class);
 
-  /**
-   * Waits for the callback to arrive for a limited amount of time.
-   * @throws InterruptedException
-   * @throws TimeoutException
-   */
-  void awaitCallback() throws InterruptedException, TimeoutException {
-    if (!latch.await(1, TimeUnit.SECONDS)) {
-      throw new TimeoutException("Waiting too long for callback to arrive");
-    }
+    // bufferList containing a null byteBuffer
+    List<ByteBuffer> bufferList = Arrays.asList(ByteBuffer.wrap(fillRandomBytes(new byte[32])), null);
```

Comment:
can you use `Utils.getRandomBytes()` and get rid of the `fillRandomBytes()` function?